aws-lc-rs/tests/cipher_test.rs

```diff
@@ -0,0 +1,360 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+use aws_lc_rs::cipher::{
+    CipherContext, DecryptingKey, EncryptingKey, OperatingMode, PaddedBlockDecryptingKey,
+    PaddedBlockEncryptingKey, PaddingStrategy, UnboundCipherKey, AES_128, AES_256,
+};
+use aws_lc_rs::iv::FixedLength;
+use aws_lc_rs::test::from_hex;
+
+macro_rules! padded_cipher_kat {
+    ($name:ident, $alg:expr, $mode:expr, $padding:expr, $constructor:ident, $key:literal, $iv: literal, $plaintext:literal, $ciphertext:literal) => { paste::item! {
+        #[test]
+        fn $name() {
+            let key = from_hex($key).unwrap();
+            let input = from_hex($plaintext).unwrap();
+            let expected_ciphertext = from_hex($ciphertext).unwrap();
+
+            let iv = from_hex($iv).unwrap();
+            let fixed_iv = FixedLength::try_from(iv.as_slice()).unwrap();
+            let context = CipherContext::Iv128(fixed_iv);
+
+            let unbound_key = UnboundCipherKey::new($alg, &key).unwrap();
+
+            let encrypting_key =
+                PaddedBlockEncryptingKey::[<less_safe_ $constructor>](unbound_key, context).unwrap();
+            assert_eq!($mode, encrypting_key.mode());
+            assert_eq!($padding, encrypting_key.padding());
+            assert_eq!($alg, encrypting_key.algorithm());
+            let mut in_out = input.clone();
+            let context = encrypting_key.encrypt(&mut in_out).unwrap();
+            assert_eq!(expected_ciphertext.as_slice(), in_out.as_slice());
+
+            let unbound_key2 = UnboundCipherKey::new($alg, &key).unwrap();
+            let decrypting_key =
+                PaddedBlockDecryptingKey::$constructor(unbound_key2, context).unwrap();
+            assert_eq!($mode, decrypting_key.mode());
+            assert_eq!($padding, decrypting_key.padding());
+            assert_eq!($alg, decrypting_key.algorithm());
+            let plaintext = decrypting_key.decrypt(&mut in_out).unwrap();
+            assert_eq!(input.as_slice(), plaintext);
```

Comment:
nit: it might be worth extracting some of this out into an actual function rather than a macro. It's especially annoying to work with `paste!` and other tooling like rustfmt or rust-analyzer.